frontend/src/components/Settings.vue

```diff
@@ -68,25 +85,87 @@
 </template>
 
 <script setup lang="ts">
+import { CheckIcon, ExclamationIcon, InformationCircleIcon } from '@heroicons/vue/solid';
+import semver from 'semver';
 import { onMounted, ref } from 'vue';
 import { useI18n } from 'vue-i18n';
 import backend, { VersionDto } from '../common/backend';
+import { FetchUpdateError, LatestVersionDto, updateChecker } from '../common/updatecheck';
 import { Locale } from '../i18n';
+import FetchError from './FetchError.vue';
 
 const { t } = useI18n({ useScope: 'global' });
 
 const version = ref<VersionDto>();
+const latestVersion = ref<LatestVersionDto>();
+const stableUpdateExists = ref<boolean>(false);
+const betaUpdateExists = ref<boolean>(false);
 const onFetchError = ref<Error | null>();
+const errorOnFetchingUpdates = ref<boolean>(false);
 
 onMounted(fetchData);
 
 async function fetchData() {
   onFetchError.value = null;
   try {
-    version.value = await backend.version.get();
+    let versionInstalled = backend.version.get();
+    let versionAvailable = updateChecker.get();
+    version.value = await versionInstalled;
+    latestVersion.value = await versionAvailable;
+    checkForUpdates();
   } catch (err) {
-    console.error('Retrieving version failed.', err);
-    onFetchError.value = err instanceof Error ? err : new Error('Unknown Error');
+    if (err instanceof FetchUpdateError) {
+      errorOnFetchingUpdates.value = true;
+    } else {
+      console.error('Retrieving version information failed.', err);
+      onFetchError.value = err instanceof Error ? err : new Error('Unknown Error');
+    }
+  }
+}
+
+function checkForUpdates() {
+  stableUpdateExists.value = checkForStableUpdate();
+  if (!stableUpdateExists.value) {
+    betaUpdateExists.value = checkForBetaUpdate();
+  }
+}
```

Comment:
I did some experiments with the `semver` lib. It is very much capable of comparing semver strings (as its name suggests). I.e.

```js
// 1.0.0 < 1.1.0-beta1 < 1.1.0-beta2 < 1.1.0:

it('1.0.0 < 1.1.0-beta1', () => {
  const result = semver.lt('1.0.0', '1.1.0-beta1');
  expect(result).to.be.true;
});

it('1.1.0-beta1 < 1.1.0-beta2', () => {
  const result = semver.lt('1.1.0-beta1', '1.1.0-beta2');
  expect(result).to.be.true;
});

it('1.1.0-beta2 < 1.1.0', () => {
  const result = semver.lt('1.1.0', '1.1.0-beta2');
  expect(result).to.be.false;
});
```

Therefore, there should be no need to manually deconstruct the version string. Regardless of beta or non-beta, just let the semver comparison do its job:

```suggestion
function checkForUpdates() {
  // TODO: can we assert that latestVersion.value != null?
  stableUpdateExists.value = semver.lt(version.value, latestVersion.value.stable);
  betaUpdateExists.value = semver.lt(version.value, latestVersion.value.beta);
}

// TODO: delete other unused functions
```

Reply:
Fixed in https://github.com/cryptomator/hub/pull/121/commits/525339bb199013a669c047e20d6571d574174fb7